spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -77,23 +77,32 @@ public DatastoreTemplate(Datastore datastore,
 	}
 
 	@Override
-	public <T> void save(T instance) {
-		this.datastore.put(convertToEntity(instance));
+	public <T> T save(T instance) {
+		Entity saved = this.datastore.put(convertToEntity(instance));
+		return this.datastoreEntityConverter.read((Class<T>) instance.getClass(), saved);
```

Comment:
This seems wasteful to do reverse conversion. Does datastore generate the ID on `put`, or do we create the ID?

Reply:
Actually I think we can just extract specifically that key from the returned Entity and set that. will change.

Reply:
done